tests/network_manager_test.go

```diff
@@ -36,3 +41,72 @@ func TestDataURLSkipRequest(t *testing.T) {
 
 	assert.True(t, lc.contains("skipped request handling of data URL"))
 }
+
+func TestMetricsEmission(t *testing.T) {
+	tb := newTestBrowser(t, withHTTPServer())
+
+	url := tb.URL("/get")
+	browserLoadedTags := map[string]string{
+		"group": "",
+		"url":   "about:blank",
+	}
+	browserTags := map[string]string{
+		"group": "",
+		"url":   url,
+	}
+	httpTags := map[string]string{
+		"method":              "GET",
+		"url":                 url,
+		"status":              "200",
+		"group":               "",
+		"proto":               "http/1.1",
+		"from_cache":          "false",
+		"from_prefetch_cache": "false",
+		"from_service_worker": "false",
+	}
+	expMetricTags := map[string]map[string]string{
+		common.BrowserDOMContentLoaded.Name:     browserLoadedTags,
+		common.BrowserLoaded.Name:               browserLoadedTags,
+		common.BrowserFirstPaint.Name:           browserTags,
+		common.BrowserFirstContentfulPaint.Name: browserTags,
+		k6metrics.DataSentName: map[string]string{
+			"group":  "",
+			"method": "GET",
+			"url":    url,
+		},
+		k6metrics.HTTPReqsName:              httpTags,
+		k6metrics.HTTPReqDurationName:       httpTags,
+		k6metrics.DataReceivedName:          httpTags,
+		k6metrics.HTTPReqConnectingName:     httpTags,
+		k6metrics.HTTPReqTLSHandshakingName: httpTags,
+		k6metrics.HTTPReqSendingName:        httpTags,
+		k6metrics.HTTPReqReceivingName:      httpTags,
```

Comment:
Should we add [other metrics here](https://pkg.go.dev/go.k6.io/k6/lib/metrics#pkg-constants) as well and check against whether they are zero or not? Are they necessary? For example, `HTTPReqFailedName`?

Reply:
These are the expected metrics, and if any unexpected ones are emitted the test will fail at
https://github.com/grafana/xk6-browser/blob/48e102154f011da7e7d2b84bd34b3e23a380046d/tests/metrics_test.go#L43

`HTTPReqFailed` is not emitted in this case.